src/EnergyPlus/SolarShading.cc

```diff
@@ -7447,7 +7447,8 @@ void CalcInteriorSolarDistribution(EnergyPlusData &state)
                                     // transmitting and back window centers
                                     CosIncBack = std::abs(state.dataBSDFWindow->ComplexWind(SurfNum).sdotN(IBack));
                                 }
-                                int const ConstrNumBackSh = state.dataSurface->SurfWinActiveShadedConstruction(BackSurfNum);
+                                int ConstrNumBackShRaw = state.dataSurface->SurfWinActiveShadedConstruction(BackSurfNum);
+                                int const ConstrNumBackSh = ConstrNumBackShRaw == 0 ? ConstrNumBack : ConstrNumBackShRaw;
```

Comment:
@jcyuan2020 This does solve the zero subscript problem, but I'm not sure this is general enough. Looking through this function, it has major blocks for the sending surface type ```
7410 if (state.dataSurface->SurfWinWindowModelType(SurfNum) == Window5DetailedModel) {
7987 } else if (state.dataSurface->SurfWinWindowModelType(SurfNum) == WindowBSDFModel) {
8146 } else if (state.dataSurface->SurfWinWindowModelType(SurfNum) == WindowEQLModel) {
```
Within each of these blocks, it appears that the back surfaces are assumed to all be of the same type as the sending surface. For example, in the first section, it always applies `PolyF` to the back surface. My guess is that this returns zero if the back surface is a complex window. For the defect file complex window I suspect that `TransBeamWin` is zero, even with this fix.

Also, line 7445 will always be false `if (state.dataSurface->SurfWinWindowModelType(SurfNum) == WindowBSDFModel) {`
because it's inside `if (state.dataSurface->SurfWinWindowModelType(SurfNum) == Window5DetailedModel) {`.

So, this fix prevents the array bounds error, but I think `CalcInteriorSolarDistribution` needs a major revision to properly handle mixed window types. Either that, or disallow mixed window types within a given enclosure when used with FullInterior solar distribution?  Not sure what's the right solution.

Reply:
@jcyuan2020 EnergyPlus models should not mix "simple" and "bsdf" window types in the same model. 
I remember problem on how to make simple window interacting with bsdf window type. One of the problems I could not solve in optical routines is how to solve interaction between them. Meaning, I could not easily solve solar absorptance in each of the layer when mixing windows. I actually didn't spend much time in trying to do mixed window models.

Reply:
Thanks @vidanovic @mjwitte for the comments! A few questions here: 1) the does the "simple" model type means a regular window (e.g. using the panes and gaps materials), or a window that uses the `WindowMaterial:SimpleGlazingSystem`? 2) to what extend should not the two types should be not mixed together: e.g. not putting these two types of windows into one zone (or space?); or not putting them into an idf file at all? 3) I am currently updating some documentation fix, so for this one would a short description about the "incompatibility" of these two window types added to the IORef helpful here? So we can add some descriptions to the complex fenestration section either here in this PR or in PR #9629.

Reply:
@jcyuan2020 No just that. In the above context simple means every window model that is not using BSDF. I would advise you not to mix them in the model at all, however, if I recall correctly, you should be good if you don't mix them in the same zones. The problem is that windows will interact if they see each other in the model. If, for example, you use a simple window model, it will calculate only one direction of the ray that is entering the zone, while the BSDF window model is calculated for more directions. The consequence is that rays that hit the back of the window models are calculated with different models which will result in incorrect calculations.

Reply:
Thanks @vidanovic ! Then probably we can limit the "incompatibility" note to a zone, and only under the scenario that the simulation uses "FullInteriorExterior" and/or "FullInteriorExteriorWithReflection" distribution calculation method. I think I remember that the code would not be activated if the "FullExterior" or "FullExteriorWithReflection" method.